feat(composer): add check for bin-dir

minorOffense opened the pull request:
Alright I'm going to preface this with this is the first time I've ever written Go code so hopefully it's not too terrible.

But this adds a method into the launcher to check composer for the bin-dir flag in the composer file. Otherwise it'll fallback to vendor/bin

minorOffense committed:
feat(composer): add check for bin-dir

dasginganinja commented:
I'm interested in bringing this one in. I think this aligns with what we're trying to do here.

Let me finish up on the `--root=` code we have in progress since we've already removed the flag module.  Once that is done I think this should merge in nicely. 

Would you be able to write some tests for this version? I might need to pitch in here, too. 😆

dasginganinja commented:
I tested this code on top of the fix in #14 and it worked as expected. 🎉 
I'm just gonna pull it in. Tests will come in time, no?

minorOffense commented:
I could take a shot at it. Looks like i'm going to need a composer.json to store in here. Suppose I should just grab the one from the core-recommended

minorOffense committed:
Merge branch 'main' into 15-composerbindir

dasginganinja commented:
We also need to make sure that situations where folks sanitize their composer.json works as well.

minorOffense commented:
Sorry what do you mean by sanitize the composer.json files? Like they remove them? Or do you mean they remove values from the json structure.

dasginganinja commented:
@minorOffense What I mean is some people may remove files like composer.json from production.  I can tell you that I remove things like install.php, .git/ and other stuff when we push to production in as part of our internal CI process. 

I'd just like to make sure this works when there is no composer.json :)  That's what the tests seemed to be complaining about, at least from what I saw.

minorOffense committed:
fix(defaults): add fallback to vendor/bin

minorOffense committed:
fix(defaults): add fallback to vendor/bin

minorOffense committed:
chore(cleanup): remove whitespace

minorOffense commented:
I think I fixed it... I can't run the tests locally. But I updated the main.go to fallback to vendor/bin if the composer finder doesn't find anything.

minorOffense committed:
fix(composer): add more fallback cases

minorOffense committed:
fix(composer): update logic

When a root is passed, we need to check the parent in case web dir is not the same a composer dir

minorOffense committed:
fix(composer): update logic

When a root is passed, we need to check the parent in case web dir is not the same a composer dir
